unit-tests/to_upper_lower.F90

```diff
@@ -0,0 +1,48 @@
+!> @file
```

Comment:
Please rename this directory from "unit_tests" to simply "tests", which is the cmake convention.